File: LoanCalculatorAPI/Models/PaymentPlan.cs
using System;

namespace LoanCalculatorAPI.Models
{
    public class PaymentPlan
    {
        public double InitialLoanAmount { get; set; }

        public double TotalPayableAmount { get; set; }

        public double MontlyRateAmount { get; set; }

        public double MonthlyInterestAmount { get; set; }

        public DateTime NextDueDate { get; set; }

        public double RemainingLoanAmount { get; set; }
    }
}

File: VismaLoanCalculator/src/app/_models/payment-plan.ts
export interface IPaymentPlan {
  initialLoanAmount: number
  totalPayableAmount: number
  montlyRateAmount: number
  monthlyInterestAmount: number
  nextDueDate: Date
  remainingLoanAmount: number
}

export class PaymentPlan implements IPaymentPlan {

  initialLoanAmount: number;
  totalPayableAmount: number;
  montlyRateAmount: number;
  monthlyInterestAmount: number;
  nextDueDate: Date;
  remainingLoanAmount: number;

  constructor() {
    this.initialLoanAmount = 10;
    this.totalPayableAmount = 0;
    this.montlyRateAmount = 0;
    this.monthlyInterestAmount = 0;
    this.nextDueDate = new Date();
    this.remainingLoanAmount = 0;
  }

}

